tools/lsp/ui/components/widgets/floating-color-picker-widget.slint

```diff
@@ -1,117 +1,677 @@
     out property <brush> picker-border-color: Palette.color-scheme == ColorScheme.dark ? #ffffff17 : transparent;
-    out property <length> left-margin: 16px;
-    out property <length> right-margin: 12px;
     out property <length> standard-margin: 16px;
+    out property <length> small-margin: 12px;
     out property <length> property-border-radius: 5px;
     out property <length> picker-width: 240px;
 }
 
-component ColorPicker {
-    in-out property <float> hue: WindowManager.current-hsv-color.hue;
-    in-out property <float> saturation: WindowManager.current-hsv-color.saturation;
-    in-out property <float> value: WindowManager.current-hsv-color.value;
-    // alpha is an int, instead of float to help snap values to whole percentage numbers
-    in-out property <int> alpha: WindowManager.current-hsv-color.alpha * 100;
-    in-out property <color> current-color: hsv(root.hue, root.saturation, root.value, root.alpha / 100);
+component FakeShadowText {
+    in property <string> text;
+    in property <string> font-family: "Inter";
+    in property <length> font-size: 11px;
+    in property <float> font-weight: 800;
+    in property <color> color: white;
 
-    in property <WidgetMode> widget-mode: edit;
+    width: t.width;
+    height: t.height;
 
-    property <length> picker-target-x;
-    property <length> picker-target-y;
+    fake-shadow := Text {
+        x: 1px;
+        y: 1px;
+        vertical-alignment: center;
+        font-family: root.font-family;
+        font-size: root.font-size;
+        color: black;
+        text: t.text;
+        font-weight: root.font-weight;
+    }
 
-    callback close <=> t-close.clicked;
+    t := Text {
+        x: 0;
+        y: 0;
+        vertical-alignment: center;
+        font-family: root.font-family;
+        font-size: root.font-size;
+        color: root.color;
+        text: root.text;
+        font-weight: root.font-weight;
+    }
 
-    changed current-color => {
-        if widget-mode == WidgetMode.preview {
-            WindowManager.update-preview-value(Api.color-to-data(root.current-color).text);
+    @children
+}
+
+component ColorIndicator {
+    width: 15px;
+    height: self.width;
+    in property <color> color;
+    property <{hue: float, saturation: float, value: float}> hsv-color: color.to-hsv();
+    Rectangle {
+        width: 100%;
+        height: 100%;
+        border-radius: 3px;
+        clip: true;
+        Rectangle {
+            x: 0;
+            width: 50%;
+            background: hsv(hsv-color.hue, hsv-color.saturation, hsv-color.value);
+        }
+        Rectangle {
+            x: parent.width / 2;
+            width: 50%;
+            background: white;
+            Image {
+                width: 100%;
+                height: 100%;
+                vertical-alignment: top;
+                horizontal-alignment: left;
+                source: @image-url("../../assets/background-tiny.svg");
```

Comment:
Should this go through the `Icons` global we have? So that we have all usages in one place if we need to replace the SVG.